File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i = 0;
        int max_profit = 0, min = 0;
        int pos = 0;
        min = prices[0];
        for (i = 0; i < prices.size(); i++) {
            if (prices[i] < min) {
                min = prices[i];
            }
            int profit = prices[i] - min;

            if ( profit > max_profit ) {
                max_profit = profit;
            }
        }
        return max_profit;
    }
};

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        max_profit = 0
        
        # Loop through the prices starting from the second day
        for i in range(1, len(prices)):
            # If the current day's price is higher than yesterday's, take the profit
            if prices[i] > prices[i - 1]:
                max_profit += prices[i] - prices[i - 1]
                
        return max_profit